classes/controller/menu/class-classic-blocks-usages.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Blocks Usage class
+ *
+ * @package P4BKS\Controllers\Menu
+ * @since 1.40.0
+ */
+
+namespace P4GBKS\Controllers\Menu;
+
+/**
+ * Show posts using classic blocks.
+ */
+class Classic_Blocks_Usages extends Controller {
+	/**
+	 * Create menu/submenu entry.
+	 */
+	public function create_admin_menu() {
+
+		$current_user = wp_get_current_user();
+
+		if ( in_array( 'administrator', $current_user->roles, true ) && current_user_can( 'manage_options' ) ) {
+			add_submenu_page(
+				P4GBKS_PLUGIN_SLUG_NAME,
+				__( 'Classic block usages', 'planet4-blocks-backend' ),
+				__( 'Classic block usages', 'planet4-blocks-backend' ),
+				'manage_options',
+				'classic_block_usages',
+				[ $this, 'classic_block_usages' ]
+			);
+		}
+	}
+
+	/**
+	 * Show all posts that use classic "blocks", which isn't really a block but the absence of one.
+	 * All consecutive content that is not inside of a block comment is regarded as a single classic block when parsed.
+	 * So in order to find classic blocks, we parse the content as blocks and then check for all blocks where the name
+	 * is null and the content is more than just whitespace.
+	 */
+	public function classic_block_usages(): void {
+		$all_posts = get_posts(
+			[
+				'post_type'   => [ 'post', 'page', 'campaign' ],
+				'numberposts' => - 1,
+			]
+		);
+
+		$has_classic_block = static function ( $post ) {
+			$blocks = parse_blocks( $post->post_content );
+			foreach ( $blocks as $block ) {
+				if ( null === $block['blockName'] && '' !== trim( $block['innerHTML'] ) ) {
+					return true;
+				}
+			}
+
+			return false;
+		};
+
+		$with_classic_blocks = array_filter( $all_posts, $has_classic_block );
+
+		// Copied this code from https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/add-classic-block-to-report/classes/controller/menu/class-blocks-usage-controller.php#L107-L107.
+		// So disabling CS here as well as this is only a temporary dev tool.
+		//phpcs:disable
+		echo '<hr>';
+		echo '<h2>Posts using classic blocks</h2>';
+		echo '<p>Following posts are detected to use classic blocks. It is recommended to convert them to new blocks.</p>';
```

Comment:
Is there a guide to convert to new blocks ? or a simple method or a url, something short we can write here to help the user ?

Reply:
Normally classic blocks should allow to click "Convert to blocks" in the block top menu, which I haven't seen failing or producing wrong output. But maybe we want to double check if this is actually safe to do in all cases before we suggest doing that.

Reply:
We could point to it, or to some documentation about [classic block](https://wordpress.org/support/article/classic-block/) and [how to convert it](https://gogutenberg.com/blocks/classic/https://gogutenberg.com/blocks/classic/), and suggest contacting us if something goes wrong ?

Reply:
Maybe that's also relevant: https://planet4.greenpeace.org/create/manage/convert-posts-to-gutenberg-editor/

Reply:
@comzeradd Great I'll add the link. How reliable is this conversion? Are there cases where it may fail, or even produce wrong content that might go unnoticed? If that's the case maybe we should warn editors to thoroughly check their content before and after?

Reply:
For the most part it is reliable. But you can never predict what kind of html will be there. You can see the content right in the editor after it's converted, before you save the post. And you can hit "Undo" on the top if you are not happy with the result. We could add that at the bottom of this handbook page.